embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/execution/HttpUrlConnectionRequestExecutionService.kt

```diff
@@ -0,0 +1,95 @@
+package io.embrace.android.embracesdk.internal.delivery.execution
+
+import io.embrace.android.embracesdk.internal.comms.api.ApiRequestV2
+import io.embrace.android.embracesdk.internal.comms.api.ApiResponse
+import io.embrace.android.embracesdk.internal.comms.api.Endpoint
+import io.embrace.android.embracesdk.internal.delivery.SupportedEnvelopeType
+import java.io.InputStream
+import java.net.HttpURLConnection
+import java.net.URL
+
+private const val NO_HTTP_RESPONSE = -1
+private const val TOO_MANY_REQUESTS = 429
+private const val DEFAULT_TIMEOUT_MILLISECONDS = 10 * 1000
+
+class HttpUrlConnectionRequestExecutionService(
+    private val coreBaseUrl: String,
+    private val lazyDeviceId: Lazy<String>,
```

Comment:
Small preference to using a provider-like interface like () -> String instead of lazy since you're not initializing a singleton, but asking a for a function to give you the value.

Same comment applies to the other implementation